tests/h/session_test.py

```diff
@@ -82,6 +83,43 @@ def test_authenticated_sidebar_tutorial(self, authenticated_request, dismissed):
             assert preferences['show_sidebar_tutorial'] is True
 
 
+class TestModelWithScopedGroups(object):
+    def test_proxies_group_lookup_to_service(self, authenticated_request):
+        svc = authenticated_request.find_service(name='list_groups')
+
+        session.model(authenticated_request)
+
+        svc.request_groups.assert_called_once_with(user=authenticated_request.user,
+                                                   authority=authenticated_request.authority)
+
+    def test_private_group_is_not_public(self, authenticated_request, factories):
+        a_group = factories.Group()
+        svc = authenticated_request.find_service(name='list_groups')
+        svc.request_groups.return_value = [a_group]
+
+        model = session.model(authenticated_request)
+
+        assert not model['groups'][0]['public']
+
+    def test_open_group_has_no_url(self, unauthenticated_request, world_group):
+        svc = unauthenticated_request.find_service(name='list_groups')
```

Comment:
It should be fine to include URLs for open groups provided they belong to the default authority.

Reply:
Ah, this is again a relic of legacy—`session` maintains its own formatting for group dicts, separate from the formatter used by the new services and API view; it has the old logic of not including a `url` for open groups. I can certainly address this but I think I'd like to do it in a separate PR as it isn't directly applicable to the current task.

Reply:
Ah, OK.